### DISCUSSION TITLE
Possible to Preserve Output File Name Even When Parameters Change?

### DISCUSSION THREAD

**Discussion by reifsnid** [doxygen/doxygen#11208] [Q&A]
Let's say I have a function with 2 parameters/arguments in an initial version of my library. Then for the next version, I add an additional parameter to the same function. The function's name stays the same.

When I run both versions of the header file through doxygen, I get two separate files for each instance of the function. However, I'm wondering if it is possible to ensure that the file name is preserved across versions for traceability. I would like users of the library to be able to easily see when the new parameter was added to the function without having to look at two separate, unique output files.

I have `SHORT_NAMES = NO` set.

**Comment by albert-github** [comment]
- which version of doxygen are you using?
- which filename is different?

I tried with a small example an compared the HTML output, an only, as expected, the contents changed

**Doxyfile**
```
QUIET =YES
SHORT_NAMES = NO
```

**Initial file**
```
/// \file

/// the fie
void fie(int arg1);
```

**Adjusted file**
```
/// \file

/// the fie
void fie(int arg1, int arg2);
````

Example with adjusted file: [example.tar.gz](https://github.com/user-attachments/files/17553397/example.tar.gz)

when you have an example where the file names change:
- please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).

**Reply by reifsnid** [reply]
Hi @albert-github,

I apologize, my original question was misleading. We are generating XML from Doxygen and then using this XML to generate our own custom HTML output for each individual member.

We base each member's custom HTML output file name on the member's id value defined in the XML generated by Doxygen. It is this id value that we've noticed changes if the member's signature/parameter list changes.

We are using version 1.8.14.

I will try to come up with an anonymized header if you still require it.

Thank you!

**Comment by albert-github** [comment]
Doxygen 1.8.14 is a bit old (December 25, 2017) the current version of doxygen is 1.12.0, so it is advised to update (won't have an effect on the issue in question though).

We are talking here about functions, initially like:
```
void fie(int arg1);
```
and later on like:
```
void fie(int arg1, int arg2);
```
there functions are in my opinion different functions and can exists both in the same code base (when overload is supported).
So the member-id is different as this reflects the signature of the function.
When you want to use the just the function name (so only possible when no use is made of overloading possibilities) you should use the field `<definition>` (and strip possibly information based on the field `<type>`).

> I will try to come up with an anonymized header if you still require it.

When the example given is not sufficient, it is always good to have:
- a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).
